### PR TITLE
GP2-127-Frontend

### PR DESCRIPTION
**Pull request by csbarbosa**
(Ticket:https://asaphub.atlassian.net/browse/GP2-127)
Delete Resource from Working Group

### CODE REVIEW

File: apps/gp2-frontend/src/working-groups/EditResourceModal.tsx [hunk 1]
```diff
@@ -40,12 +40,17 @@ const EditResourceModal: React.FC<EditResponseModalProps> = ({
...
+        updateWorkingGroupResources(
+          Object.assign([], resources, {
             [routeIndex]: resource,
           }),
-        ]);
+        )
+      }
+      onDelete={() => {
+        const newResources = [...resources];
+        newResources.splice(routeIndex, 1);
```
Comment (markcsims) [inline apps/gp2-frontend/src/working-groups/EditResourceModal.tsx:52]:
```suggestion
        const newResources = [...resources]
          .splice(routeIndex, 1);
```

File: packages/gp2-components/src/organisms/ResourceModal.tsx [hunk 2]
```diff
@@ -42,7 +52,8 @@ type ResourceModalProps = Partial<gp2.Resource> & {
   modalTitle: string;
   modalDescription: string;
   backHref: string;
-  onSave?: (data: gp2.Resource) => void | Promise<void>;
+  onSave?: (data: gp2.Resource) => Promise<void>;
+  onDelete?: () => void | Promise<void>;
```
Comment (markcsims) [inline packages/gp2-components/src/organisms/ResourceModal.tsx:56]:
```suggestion
  onDelete?: () => Promise<void>;
```

File: packages/gp2-components/src/organisms/__tests__/ResourceModal.test.tsx [hunk 3]
```diff
@@ -132,6 +138,24 @@ describe('ResourceModal', () => {
...
+    });
+    it('allows a resource to be deleted', async () => {
+      const onDelete = jest.fn();
+      renderResourseModal({ onDelete });
+      enterType('Link');
+      enterTitle('a title');
+      enterLink('http://example.com');
+      const deleteButton = deleteClick();
+      await waitFor(() => expect(deleteButton).toBeEnabled());
+      expect(onDelete).toBeCalledWith();
```
Comment (markcsims) [inline packages/gp2-components/src/organisms/__tests__/ResourceModal.test.tsx:157]: If you want to check the method is called use toBeCalled.

Or do you want to check what onDelete has been called with?

File: packages/react-components/src/organisms/EditModal.tsx [hunk 4]
```diff
@@ -56,14 +56,14 @@ const EditModal: React.FC<EditModalProps> = ({
   }, [status, dirty]);
 
   const historyPush = usePushFromHere();
-  const asyncOnSave = async () => {
+
+  const asyncFunctionWrapper = async (cb: () => void | Promise<void>) => {
```
Comment (markcsims) [inline packages/react-components/src/organisms/EditModal.tsx:60]:
```suggestion
  const asyncFunctionWrapper = async (cb: () => Promise<void>) => {
```

Should this function be returning a Promise only?

@csbarbosa  @jhonny-m

File: packages/react-components/src/organisms/EditModal.tsx [hunk 5]
```diff
@@ -32,7 +32,7 @@ type EditModalProps = Pick<
   dirty: boolean; // mandatory so that it cannot be forgotten
   children: (
     state: { isSaving: boolean },
-    asyncOnSave: () => void,
+    asyncFunctionWrapper: (cb: () => void | Promise<void>) => void,
```
Comment (markcsims) [inline packages/react-components/src/organisms/EditModal.tsx:35]:
```suggestion
    asyncFunctionWrapper: (cb: () => Promise<void>) => void,
```

see below

File: apps/gp2-frontend/src/working-groups/EditResourceModal.tsx (hunk 1 above)
Comment (csbarbosa) [reply apps/gp2-frontend/src/working-groups/EditResourceModal.tsx:52]: Done

File: packages/gp2-components/src/organisms/__tests__/ResourceModal.test.tsx (hunk 3 above)
Comment (csbarbosa) [reply packages/gp2-components/src/organisms/__tests__/ResourceModal.test.tsx:157]: Clarified

**Review by markcsims — APPROVED** [review]

**Pull request merged**